### DISCUSSION TITLE
Is it possible to increase ram after creating the container?

### DISCUSSION THREAD

**Discussion by marcocastignoli** [sickcodes/Docker-OSX#199] [Q&A]
Is it possible to increase ram after creating the container?

**Comment by w84death** [comment]
I got here with the same problem. I have 32GB to use and the container only senses 2GB.

**Comment by sickcodes** [comment]
I'll write a script later today

**Reply by sickcodes** [reply]
The easiest way would be `docker start -ai yourcontainer`

`docker exec -it containerid /bin/bash`

`vi Launch.sh`

And add `RAM=16` for 16GB of ram at the top of the script

**Reply by sickcodes** [reply]
The correct solution is: https://github.com/sickcodes/Docker-OSX#quick-start-own-image-naked-container-image

**Comment by avijoni** [comment]
docker update --memory=16g container-name

**Comment by sickcodes** [comment]
> docker update --memory=16g container-name
> 
> 

Woah didn't even know you could do Docker update!

**Reply by avijoni** [reply]
Btw, don't forget to expand swap space. I had issue adding more ram, without adding more swap space. Swap should be at least the size of memory. Hope this helps.

docker update --memory=4g --memory-swap=4g container-name

you can put --memory-swap=-1 for unlimited 

Cheers.

**Comment by rafaseron** [comment]
im tryng to change ram memory but no luck
i changed the launch.sh and the docker ram, but macos does not recognize more than 4gb of ram (my windows pc is 32gb and wsl gives full access to windows ram, does not limit)
can you help me where exactly i change to recognize more ram in macos?
more info: im running docker-osx:ventura
*I would like to use macos ventura to study development on xcode*

**Reply by MitjaHenner** [reply]
Having the same problem.
EDIT:
Recreated the container with RAM=16

**Reply by artschur** [reply]
> Having the same problem.
> 
> EDIT:
> 
> Recreated the container with RAM=16

How were you able to change the ram on it??

**Comment by jos-blue-ocean-robotics-com** [comment]
I had the most success following this:
> The easiest way would be `docker start -ai yourcontainer`
> 
> `docker exec -it containerid /bin/bash`
> 
> `vi Launch.sh`
> 
> And add `RAM=16` for 16GB of ram at the top of the script

When adding more cpu I had to add the following to lines in the beginning of Launch.sh  like this:
```
CORES=8
SMP=8
```
Just adding CORES does not work and the container exits immediately after starting.

**Reply by mateusdata** [reply]
Realmente tive sucesso aqui. muito obrigada. Depois de 3 dias tentando  consequi agora.

I really had success here. thank you very much. After 3 days of trying I got it now.

**Reply by antoniovini47** [reply]
![image](https://github.com/user-attachments/assets/511284d8-87c0-4882-b587-ecc23b5bd7dd)

Thanks, that worked! Just need to restart the container after that.

**Reply by mateusdata** [reply]
Perfeito.

**Reply by dannypurcell** [reply]
Brilliant, Thanks!

Confirming this works, just went through the following:

* Set up a Sonoma image according to the [README](https://github.com/sickcodes/Docker-OSX?tab=readme-ov-file#sonoma-14-)
* Run `docker update --memory=4g --memory-swap=4g <container-name>` 
* `docker exec -it <container-name> /bin/bash`
* Make the top of file edits as shown in the screenshot here, save and exit vi
* Shutdown from OSX Apple menu
* `docker restart <container-name>`

Now able to run XCode Simulator to run on a Sonoma 14.7.2 image
